File: shared/contracts/python/draft_claim_note.py
"""
DraftClaimNote contract — Python / Pydantic.
Represents an AI-generated note draft awaiting human review and approval.
"""

from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel

from .evidence_source import EvidenceSource
from .governance_decision import GovernanceDecision
from .model_route_decision import ModelRouteDecision


class DraftNoteStatus(str, Enum):
    PENDING_REVIEW = "PENDING_REVIEW"
    APPROVED = "APPROVED"
    REJECTED = "REJECTED"
    WRITTEN = "WRITTEN"
    WRITE_FAILED = "WRITE_FAILED"


class NoteCategory(str, Enum):
    PROGRESS_NOTE = "PROGRESS_NOTE"
    COVERAGE_ANALYSIS = "COVERAGE_ANALYSIS"
    LIABILITY_ASSESSMENT = "LIABILITY_ASSESSMENT"
    RESERVE_JUSTIFICATION = "RESERVE_JUSTIFICATION"
    CLOSING_NOTE = "CLOSING_NOTE"
    ESCALATION_NOTE = "ESCALATION_NOTE"


class DraftClaimNote(BaseModel):
    draft_id: str
    claim_id: str
    conversation_id: str | None = None
    status: DraftNoteStatus
    category: NoteCategory
    created_at: datetime
    updated_at: datetime

    # AI-generated content
    ai_content: str

    # Human-edited content (starts as copy of ai_content)
    edited_content: str

    # Grounding
    evidence_sources: list[EvidenceSource] = []

    # Governance — always ESCALATE for draft_note task type
    governance_decision: GovernanceDecision
    model_route_decision: ModelRouteDecision

    # Human decision
    reviewed_by: str | None = None
    reviewed_at: datetime | None = None
    reviewer_comments: str | None = None

    # Write outcome
    claim_center_note_id: str | None = None
    write_attempted_at: datetime | None = None
    write_failure_reason: str | None = None

File: shared/contracts/typescript/draftClaimNote.ts
// DraftClaimNote contract
// Represents an AI-generated note draft awaiting human review and approval.

import type { EvidenceSource } from "./evidenceSource";
import type { GovernanceDecision } from "./governanceDecision";
import type { ModelRouteDecision } from "./modelRouteDecision";

export enum DraftNoteStatus {
  PendingReview = "PENDING_REVIEW",   // Awaiting human review
  Approved = "APPROVED",              // Human approved — ready to write
  Rejected = "REJECTED",              // Human rejected — draft discarded
  Written = "WRITTEN",                // Successfully written to ClaimCenter
  WriteFailed = "WRITE_FAILED",       // Write attempted but failed
}

export enum NoteCategory {
  ProgressNote = "PROGRESS_NOTE",
  CoverageAnalysis = "COVERAGE_ANALYSIS",
  LiabilityAssessment = "LIABILITY_ASSESSMENT",
  ReserveJustification = "RESERVE_JUSTIFICATION",
  ClosingNote = "CLOSING_NOTE",
  EscalationNote = "ESCALATION_NOTE",
}

export interface DraftClaimNote {
  draftId: string;
  claimId: string;
  conversationId: string | null;  // originating conversation, if applicable
  status: DraftNoteStatus;
  category: NoteCategory;
  createdAt: string;              // ISO 8601
  updatedAt: string;

  // AI-generated content
  aiContent: string;

  // Human-edited content (starts as copy of aiContent, user may edit before approving)
  editedContent: string;

  // Grounding
  evidenceSources: EvidenceSource[];

  // Governance — always ESCALATE for draft_note task type
  governanceDecision: GovernanceDecision;
  modelRouteDecision: ModelRouteDecision;

  // Human decision
  reviewedBy: string | null;
  reviewedAt: string | null;      // ISO 8601
  reviewerComments: string | null;

  // Write outcome (populated after status = WRITTEN or WRITE_FAILED)
  claimCenterNoteId: string | null;
  writeAttemptedAt: string | null;
  writeFailureReason: string | null;
}
